dlt/common/schema/utils.py

```diff
@@ -240,12 +246,18 @@ def compile_simple_regexes(r: Iterable[TSimpleRegex]) -> REPattern:
 
 
 def validate_stored_schema(stored_schema: TStoredSchema) -> None:
+    # exclude validation of keys added later
+    ignored_keys = []
```

Comment:
why this is required? first we migrate the dictionary and then we validate the schema. so the engine should be always 7

Reply:
in test_upgrade_engine_v1_schema many different schema versions are validated. we could alternatively change that test.

Reply:
really I was sure we migrate and then validate. We should validate only the version after migration. So you can change the test.

but you can keep this `filter_required` providing that you add tests for it